src/braket/annealing/problem.py

```diff
@@ -37,16 +37,16 @@ class Problem:
     def __init__(
         self,
         problem_type: ProblemType,
-        linear: Dict[int, float] = None,
-        quadratic: Dict[Tuple[int, int], float] = None,
+        linear: Dict[int, float] | None = None,
```

Comment:
```suggestion
        linear: dict[int, float] | None = None,
```

Python supports `dict`, `list`, `tuple` as a built in type now. This can also be removed from imports.

Reply:
Yes, but I'd prefer not to make those types of changes in this PR (this PR is broad enough without them).

Reply:
Maybe we can add a rule to the linter that flags imports of `Dict`, `List`, etc from `typing`